### PR TITLE
[ci] migrate the rest of cpp tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Migrate tsan, ubsan and core worker tests to civ2

Test:
- CI

### CODE REVIEW

File: .buildkite/core.rayci.yml [hunk 1]
```diff
@@ -184,6 +184,16 @@ steps:
       - "3.10"
       - "3.11"
 
+  # cpp tests
+  - label: ":ray: core: cpp worker tests"
+    tags: core_cpp
+    instance_type: small
+    commands:
+      - ci/ci.sh build
+      - ci/ci.sh test_cpp
```
Comment (aslonnie) [inline .buildkite/core.rayci.yml:193]: do we have a plan to stop using this `ci/ci.sh` script in the future?

File: BUILD.bazel [hunk 2]
```diff
@@ -1326,7 +1326,7 @@ ray_cc_test(
     name = "worker_pool_test",
     size = "small",
     srcs = ["src/ray/raylet/worker_pool_test.cc"],
-    tags = ["team:core"],
+    tags = ["team:core", "no_tsan"],
```
Comment (aslonnie) [inline BUILD.bazel:1329]: do we know why does tsan do not work for these tests? is it by design or there are bugs preventing them to pass tsan?

Comment (rkooo567) [reply BUILD.bazel:1329]: cc @scv119 do you remember?

File: BUILD.bazel [hunk 3]
```diff
@@ -1486,6 +1486,7 @@ ray_cc_test(
     tags = [
         "stats",
         "team:core",
+        "no_tsan",
```
Comment (rkooo567) [inline BUILD.bazel:1489]: Q: this will keep the status quo right? (we will still run tests with tsan)

**Review by rkooo567 — APPROVED** [review]

File: BUILD.bazel (hunk 3 above)
Comment (can-anyscale) [reply BUILD.bazel:1489]: Yess, it's the same behavior as before

File: .buildkite/core.rayci.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/core.rayci.yml:193]: Yess, there are only a few straggler remained

File: BUILD.bazel (hunk 2 above)
Comment (can-anyscale) [reply BUILD.bazel:1329]: I'm not exactly sure, but these tests do not pass. Skipping these tests is the same behavior as before though.

**Comment by can-anyscale** [comment]
Merging now, folks feel free to discuss why some tests need to be omitted and we can adjust as a follow up.

**Pull request merged**